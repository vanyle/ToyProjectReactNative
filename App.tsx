import React from "react";
import type { Node } from "react";


import data from "./dataTest.json"
import {
  SafeAreaView,
  ScrollView,
  StatusBar,
  useColorScheme,
  View,
  Text,
  FlatList,
} from "react-native";

import {
  Colors,
} from "react-native/Libraries/NewAppScreen";
import Cocktail from "./cocktail";

const App: () => Node = () => {
  const isDarkMode = useColorScheme() === "dark";

  const backgroundStyle = {
    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
   };

  return (
    <SafeAreaView style={backgroundStyle}>
      <StatusBar
        barStyle={isDarkMode ? "light-content" : "dark-content"}
        backgroundColor={backgroundStyle.backgroundColor}
      />
      <ScrollView
        contentInsetAdjustmentBehavior="automatic"
        style={backgroundStyle}
      >
      <View
          style={{
            backgroundColor: isDarkMode ? Colors.black : Colors.white,
          }}
        >
        <View>
        <Text style={{fontSize: 40}}>Welcome to the Cocktail app!</Text>
        </View>
        <FlatList data={data.drinks}
        renderItem={({item}) => {
          return (<Cocktail name={item.strDrink} description={item.strInstructions} />)
        }}>
        </FlatList>
        </View>
      </ScrollView>
    </SafeAreaView>
  );
};


export default App;
