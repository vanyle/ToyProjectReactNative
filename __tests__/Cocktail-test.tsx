/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../cocktail';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import Cocktail from '../cocktail';

it('renders correctly', () => {
  const tree = renderer.create(<Cocktail />).toJSON();
  expect(tree).toMatchSnapshot();
});
