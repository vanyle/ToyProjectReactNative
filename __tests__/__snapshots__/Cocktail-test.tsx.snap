// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "red",
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "green",
      }
    }
  >
    Vodka, Orange Juice and Cointreau
  </Text>
</View>
`;
